tests/functional/ArchiveTest.php

```diff
@@ -328,9 +344,24 @@ public function testArchiveRestoreCommand(): void
             'Database connection settings are required if --destination-path',
             $this->getErrorOutput()
         );
+        $this->installComposerDependencies();
+        $this->addDrushViaPathRepository();
         $this->assertRestoredSiteStatus();
     }
 
+    private function addDrushViaPathRepository(): void
+    {
+        // When the Drush SUT is archived, it does not include Drush itself,
+        // because Drush is the project root, and therefore does not appear in the
+        // composer.json of the archived Drupal site. Drush 12.x+ cannot bootstrap
+        // any site other than the one it shares its vendor directory with. We
+        // must therefore add Drush via a Composer PATH repository in the site
+        // created via archive restore before we can run Drush commands on it.
+        $this->execOnRestoredSite(['composer', 'config', 'repositories.drush', 'path', dirname(self::getDrush())]);
```

Comment:
This is editing the composer.json in our root? If so does the test reliably undo the changes?

Reply:
This is editing the composer.json on a copy / fixture Drupal site, not the SUT. The fixture is thrown away after the test.